pkg/gke/components/Networking.vue

```diff
@@ -0,0 +1,771 @@
+      } else {
+        out.push(...(this.subnetworks.filter((s) => s.network.split('/').pop() === this.network) || []));
+      }
+
+      const labeled = out.map((sn: GKESubnetwork) => {
+        const name = sn.name ? sn.name : (sn.subnetwork || '').split('/').pop();
+
+        return {
+          name, label: `${ name } (${ sn.ipCidrRange })`, ...sn
+        };
+      });
+
+      if (this.useIpAliases) {
+        labeled.unshift({ label: this.t('gke.subnetwork.auto'), name: GKE_NONE_OPTION });
+      }
+
+      return labeled;
+    },
+
+    clusterSecondaryRangeOptions(): {rangeName: string, ipCidrRange?: string, label: string}[] {
+      if (this.selectedSubnetwork && this.selectedSubnetwork.name === GKE_NONE_OPTION) {
+        return [{
+          label:     this.t('generic.none'),
+          rangeName: GKE_NONE_OPTION
+        }];
+      }
+
+      const out: {rangeName: string, ipCidrRange?: string, label: string}[] = (this.selectedSubnetwork?.secondaryIpRanges || []).map(({ ipCidrRange, rangeName }) => {
+        return {
+          rangeName,
+          ipCidrRange,
+          label: `${ rangeName } (${ ipCidrRange })`
+        };
+      });
+
+      out.unshift({
+        label:     this.t('generic.none'),
+        rangeName: GKE_NONE_OPTION
+      });
+
+      return out;
+    },
+    /**
+     * Only the user-defined network and subnetwork names appear in the GKE config
+     * selectedNetwork and selectedSubnetwork keep track of all the additional networking info from gcp api calls
+     * eg subnets' ipCidrRange, to display alongside name in the dropdown
+     */
+    selectedNetwork: {
+      get() {
+        const { network } = this;
+
+        if (!network) {
+          return undefined;
+        }
+
+        return this.networkOptions.find((n) => n.name === network);
+      },
+      set(neu:{name:string}) {
+        this.$emit('update:network', neu.name);
+      }
+    },
+
+    selectedSubnetwork: {
+      get(): {label: string, name: string, secondaryIpRanges?: any[]} | undefined {
+        const { subnetwork } = this;
+
+        if (!subnetwork || subnetwork === '') {
+          return { label: this.t('gke.subnetwork.auto'), name: GKE_NONE_OPTION };
+        }
+
+        return this.subnetworkOptions.find((n) => n.name === subnetwork);
+      },
+      set(neu:{name:string}) {
+        if (neu.name === GKE_NONE_OPTION) {
+          this.$emit('update:subnetwork', '');
+        } else {
+          this.$emit('update:subnetwork', neu.name);
+        }
+      }
+    },
+
+    selectedClusterSecondaryRangeName: {
+      get(): {rangeName: string, ipCidrRange?: string, label: string} | undefined {
+        if (!this.clusterSecondaryRangeName) {
+          return {
+            label:     this.t('generic.none'),
+            rangeName: GKE_NONE_OPTION
+          };
+        } else return this.clusterSecondaryRangeOptions.find((opt) => opt.rangeName === this.clusterSecondaryRangeName);
```

Comment:
nitpick, code style wise we always wrap in `{ }`. In this case though it could just be
```
if (this.clusterSecondaryRangeOptions) {
   return this.clusterSecondaryRangeOptions.find
}

return {
...
}
```

or super safe and simpler
```
        return this.clusterSecondaryRangeOptions.find((opt) => opt.rangeName === this.clusterSecondaryRangeName) || {
          label:     this.t('generic.none'),
          rangeName: GKE_NONE_OPTION
        };
```

Reply:
Yeah, that is a little sloppy looking. Logic tweaked slightly from what you suggested because of some changes I made to accomodate the view config mode, but I think I covered the gist of what you're saying.